### ISSUE TITLE
unnecessary_parenthesis false negative

### ISSUE THREAD

**Issue by chunhtai** [dart-lang/linter#4453]
See the code here

https://github.com/flutter/flutter/blob/ca11492b6bdc0f49708173324dccc461fc20e1d9/packages/flutter/lib/src/material/segmented_button.dart#L343

```dart
      final Widget buttonWithTooltip = (segment.tooltip != null)
        ? Tooltip(
            message: segment.tooltip,
            child: button,
          )
        : button;
```

the parentheses should be dropped but aren't flagged by the lint

**Comment by srawlins** [comment]
This is intentional. In an expression with whitespace between certain tokens, parens can improve readability. See the lint [description](https://dart.dev/tools/linter-rules/unnecessary_parenthesis) and this test case in the tests:

`(a ?? true) ? true : true; // OK`
